crates/node/src/lib.rs

```diff
@@ -439,11 +439,12 @@ async fn handle_line(node: &mut Node, line: String) -> eyre::Result<()> {
                             last_transaction_hash: calimero_primitives::hash::Hash::default(),
                         };
 
+                        // We don't have the identity at this point
                         node.ctx_manager
                             .add_context(
                                 context,
-                                calimero_primitives::identity::ContextIdentity {
-                                    public_key: *context_id,
+                                KeyPair {
+                                    public_key: PublicKey(*context_id),
```

Comment:
the same way the user of the interactive CLI provides a context ID, they can provide a private key for their executor (from which we can extrapolate a `KeyPair`)

Reply:
That makes a lot of sense. Thanks - that clears up where to get this from 🙂 

I have added the parameter along with extraction of the key and some `From` conversions for it.